src/Abstractions/NexusMods.Abstractions.Jobs/AJobWorker.cs

```diff
@@ -0,0 +1,176 @@
+        {
+            job.SetWorker(this);
+            return;
+        }
+
+        Debug.Assert(ReferenceEquals(job.Worker, this), "same worker");
+    }
+
+    /// <inheritdoc/>
+    public ValueTask StartAsync(IJob input, CancellationToken cancellationToken = default)
+    {
+        if (input is not AJob job) throw new NotSupportedException();
+        cancellationToken.ThrowIfCancellationRequested();
+
+        SetWorker(job);
+        if (job.Status == JobStatus.None)
+        {
+            job.SetStatus(JobStatus.Created);
+        }
+
+        if (job.Status == JobStatus.Paused)
+        {
+            Debug.Assert(job.Task is null || job.Task.IsCompleted, "task should've completed");
+            job.Task = null;
+            var didReset = job.CancellationTokenSource.TryReset();
+            Debug.Assert(didReset, "reset should work");
+        }
+
+        job.SetStatus(JobStatus.Running);
+
+        Task<OneOf<JobResult, Paused>> task;
+        try
+        {
+            task = Task.Run(async () =>
+            {
+                var result = await ExecuteAsyncWrapper(job, job.CancellationTokenSource.Token);
+                if (result.IsT0)
+                {
+                    job.SetResult(result.AsT0, inferStatus: true);
+                }
+                else
+                {
+                    job.SetStatus(JobStatus.Paused);
+                }
+
+                job.Task = null;
+                return result;
+            }, cancellationToken: CancellationToken.None);
+        }
+        catch (Exception e)
+        {
+            // synchronous or async startup exception
+            job.SetResult(JobResult.CreateFailed(e), inferStatus: true);
+            return ValueTask.CompletedTask;
+        }
+
+        job.Task = task;
+        return ValueTask.CompletedTask;
+    }
+
+    private static JobResult TaskResultToJobResult(Task<JobResult> task)
+    {
+        if (task.IsCompletedSuccessfully)
+        {
+            return task.Result;
+        }
+
+        if (task.IsCanceled)
+        {
+            return JobResult.CreateCancelled();
+        }
+
+        if (task.IsFaulted)
+        {
+            return JobResult.CreateFailed(task.Exception);
+        }
+
+        throw new UnreachableException();
+    }
+
+    /// <inheritdoc/>
+    public async ValueTask PauseAsync(IJob input, CancellationToken cancellationToken = default)
+    {
+        if (input is not AJob job) throw new NotSupportedException();
+        cancellationToken.ThrowIfCancellationRequested();
+        job.Status.AssertTransition(JobStatus.Paused);
+
+        var tsc = new TaskCompletionSource();
+        using var disposable = job.ObservableStatus.Subscribe(status =>
+        {
+            if (status == JobStatus.Paused) tsc.SetResult();
+        });
+
+        job.IsRequestingPause = true;
+        await job.CancellationTokenSource.CancelAsync();
+        await tsc.Task.WaitAsync(cancellationToken: cancellationToken);
```

Comment:
This code template to await the change of the status is used multiple times, I would suggest the following to simplify it:

```suggestion
        var statusUpdate = job.ObservableStatus
            .FirstAsync(status => status == JobStatus.Paused)
            .ToTask(cancellationToken);

        job.IsRequestingPause = true;
        await job.CancellationTokenSource.CancelAsync();
        await statusUpdate;
```